cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java

```diff
@@ -65,22 +65,24 @@
   protected ExecutorService _metricProcessorExecutor;
   protected String _partitionMetricSampleStoreTopic;
   protected String _brokerMetricSampleStoreTopic;
+  protected Integer _metricSampleStoreTopicReplicationFactor;
   protected volatile double _loadingProgress;
   protected Producer<byte[], byte[]> _producer;
   protected volatile boolean _shutdown = false;
 
+  public static final String PARTITION_METRIC_SAMPLE_STORE_TOPIC_CONFIG = "partition.metric.sample.store.topic";
+  public static final String BROKER_METRIC_SAMPLE_STORE_TOPIC_CONFIG = "broker.metric.sample.store.topic";
+  public static final String NUM_SAMPLE_LOADING_THREADS_CONFIG = "num.sample.loading.threads";
+  public static final String NUM_METRIC_SAMPLE_STORE_TOPIC_REPLICATION_FACTOR_CONFIG = "num.metric.sample.store.topic.replication.factor";
```

Comment:
It looks like we lost the default values for the configs copied from `KafkaCruiseControlConfig.java`. Can we ensure that the default values are specified here as well? -- e.g. for `PARTITION_METRIC_SAMPLE_STORE_TOPIC_CONFIG`, the default value is "__KafkaCruiseControlPartitionMetricSamples".

Reply:
for `PARTITION_METRIC_SAMPLE_STORE_TOPIC_CONFIG` and `BROKER_METRIC_SAMPLE_STORE_TOPIC_CONFIG` we use `KafkaCruiseControlUtils.getRequiredConfig` which will throw exception if user does not explicitly specify the topic name while they are using this implementation of `SampleStore`. I feel this is reasonable. 

for `NUM_SAMPLE_LOADING_THREADS_CONFIG` we have the default value(set to 8). 

for `NUM_METRIC_SAMPLE_STORE_TOPIC_REPLICATION_FACTOR_CONFIG`, it is set to `Math.min(2, numberOfBrokersInCluster)`.

I update the JavaDoc of this class to mention these are required configs. Do you think that is clear enough?

Reply:
Sure, looks good to me!
-- let's make sure that we mention this behavior change in commit message.